src/getVariants.ts

```diff
@@ -0,0 +1,108 @@
+import { AtRule, Container, Node } from 'postcss';
+
+import { JitState } from '..';
+
+function isAtRule(node: Node): node is AtRule {
+  return node.type === 'atrule';
+}
+
+export default function getVariants(state: JitState): Record<string, string | null> {
+  function escape(className: string): string {
+    const node = state.modules.postcssSelectorParser.module.className({ value: className });
+    return node.value;
+  }
+
+  const result: Record<string, string | null> = {};
+  // [name, [sort, fn]]
+  // [name, [[sort, fn]]]
+  for (const [variantName, variantFnOrFns] of state.jitContext.variantMap) {
+    const fns = (Array.isArray(variantFnOrFns[0]) ? variantFnOrFns : [variantFnOrFns]).map(
```

Comment:
i could not get this to work with type inference ...

Reply:
I’ll have a look :+1:

Reply:
even their inline comment was wrong - the data dindt match 😂